From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gaurav Kashyap <gaurkash@codeaurora.org>
Date: Fri, 22 May 2020 13:45:18 -0700
Subject: ANDROID: scsi: ufs: Handle clocks when lrbp fails

Release UFS clocks when ufshcd_prepare_lrbp_crypto fails
for any reason.

Bug: 157284579
Change-Id: I32128709fb3b653ad5a2aa8d9b680c34a81f667f
Signed-off-by: Gaurav Kashyap <gaurkash@codeaurora.org>
---
 drivers/scsi/ufs/ufshcd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/scsi/ufs/ufshcd.c b/drivers/scsi/ufs/ufshcd.c
--- a/drivers/scsi/ufs/ufshcd.c
+++ b/drivers/scsi/ufs/ufshcd.c
@@ -2532,6 +2532,7 @@ static int ufshcd_queuecommand(struct Scsi_Host *host, struct scsi_cmnd *cmd)
 	err = ufshcd_prepare_lrbp_crypto(hba, cmd, lrbp);
 	if (err) {
 		lrbp->cmd = NULL;
+		ufshcd_release(hba);
 		goto out;
 	}
 	lrbp->req_abort_skip = false;
